drizzlepac/devutils/search_skyframes.py

```diff
@@ -0,0 +1,323 @@
+                             "specified, columns will be displayed in the query results.")
+    parser.add_argument('--output_sorting', required=False, default="None", nargs='?',
+                        help="Order (if any) in which to sort columns of the query results. Needs to be some "
+                             "combination of 'dateobs', 'exposure', 'skycell', 'config' and/or 'spec'. If "
+                             "not explicitly specified, query results will not be sorted. Recommended "
+                             "sorting order is 'skycell,config,spec,exposure'.")
+    in_args = parser.parse_args()
+
+    arg_dict = {}
+    for item in in_args.__dict__.keys():
+        if item not in ["master_observations_file", "output_results_file",
+                        "output_columns", "output_sorting", "visualize_footprints"]:
+            arg_dict[item] = in_args.__dict__[item]
+
+    # Reformat input args
+    blank_entry_ctr = 0
+    for item in arg_dict.keys():
+        if arg_dict[item] == "None":
+            arg_dict[item] = None
+            blank_entry_ctr += 1
+
+    if arg_dict['spec']:
+        arg_dict['spec'] = arg_dict['spec'].lower()
+
+    if in_args.output_results_file == "None":
+        in_args.output_results_file = None
+
+    if in_args.output_columns == "None":
+        in_args.output_columns = None
+    else:
+        in_args.output_columns = in_args.output_columns.split(",")
+
+    if in_args.output_sorting == "None":
+        in_args.output_sorting = None
+    else:
+        in_args.output_sorting = in_args.output_sorting.split(",")
+
+    # FAULT TOLERANCE. Exit if something in the input arguments isn't quite right.
+    # Exit if user didn't enter any search criteria
+    if blank_entry_ctr == 5:
+        sys.exit("ERROR: Search results too broad. No search criteria were entered.")
+    # Exit if specified dates don't match the format "YYYY-MM-DD".
+    if arg_dict["date_range"]:
+        date_range_dt = []
+        for date_string, date_label in zip(arg_dict["date_range"], ["Start", "End"]):
+            try:
+                date_range_dt.append(datetime.strptime(date_string, "%Y-%m-%d"))
+            except ValueError:
+                sys.exit("{} date value '{}' is not properly formatted. Please use the format 'YYYY-MM-DD'.".format(date_label, date_string))
+        # flip order of dates if first is later than the second.
+        if date_range_dt[1] < date_range_dt[0]:
+            arg_dict["date_range"].reverse()
+
+    # Exit if in_args.master_observations_file == in_args.output_results_file so the master observations table doesn't get overwritten
+    if in_args.master_observations_file == in_args.output_results_file:
+        sys.exit("ERROR: The output results file cannot have the same name is the input master observations table")
+
+    # Exit if in_args.master_observations_file is not found.
+    if in_args.master_observations_file is "":
```

Comment:
It may also be more direct, and less prone to misinterpretation, to simply check for the existence of `ALL_EXP_FILE` in `os.environ` and check that it exists.